### PR TITLE
feature: added update course. Refactored course attributes, refactore…

### LINKED ISSUE #90: C-3-1-a-EditCourseDraft

### LINKED ISSUE #91: c-3-1-b-SubmitCourseForApproval
If sensei is not approved, return 'Your profile has not been verified by diji dojo yet'. If approved, change state to pending

### PR DESCRIPTION
**Pull request by kengthong**
### Changelog:
- Added update course. 
- Refactored course attributes
- Refactored get user profile return response

## Description:
_Eg; Fixed router bug_
## Checklist:
- [x] Merged latest develop
- [x] PR title makes sense

### CODE REVIEW

File: src/controllers/course.controller.ts [hunk 1]
```diff
@@ -29,7 +29,42 @@ export class CourseController {
     }
   }
 
-  // ==================== COURSE CONTRACT ====================
+  public static async updateCourse(req, res) {
+    const { user } = req;
+    const { courseId } = req.params;
+    console.log('courseId =', courseId);
```
Comment (ianleck) [inline src/controllers/course.controller.ts:35]: rmb remove next time you touch this file

File: src/models/Course.ts [hunk 2]
```diff
@@ -32,12 +32,24 @@ export class Course extends BaseEntity {
...
+
+  @AllowNull(true)
+  @Column(DataType.TEXT)
+  subTitle: string;
 
   @AllowNull(false)
   @Column(DataType.TEXT)
   description: string;
 
+  @AllowNull(true)
```
Comment (ianleck) [inline src/models/Course.ts:45]: if allowNull is true then don't need to put I believe

File: src/models/Course.ts [hunk 3]
```diff
@@ -32,12 +32,24 @@ export class Course extends BaseEntity {
 
   @AllowNull(false)
   @Column(DataType.STRING)
-  name: string;
+  title: string;
+
+  @AllowNull(true)
```
Comment (ianleck) [inline src/models/Course.ts:37]: if allowNull is by default true I think? Not sure if still need the decorator at all

File: src/models/Course.ts [hunk 4]
```diff
@@ -48,11 +60,6 @@ export class Course extends BaseEntity {
   @Column(DataType.STRING)
   currency: string;
 
-  @AllowNull(true)
-  @Default(null)
-  @Column(DataType.STRING)
-  parentCourseId: string;
-
   @AllowNull(true)
```
Comment (ianleck) [inline src/models/Course.ts:63]: ^

File: src/models/Course.ts (hunk 3 above)
Comment (ianleck) [reply src/models/Course.ts:37]: if still need decorator can just put `@AllowNull` without the `(true)`

File: src/controllers/course.controller.ts (hunk 1 above)
Comment (ianleck) [inline src/controllers/course.controller.ts:35]: remove console log

**Review by ianleck — CHANGES_REQUESTED** [review]

File: src/controllers/course.controller.ts (hunk 1 above)
Comment (kengthong) [reply src/controllers/course.controller.ts:35]: ok

File: src/models/Course.ts (hunk 2 above)
Comment (kengthong) [reply src/models/Course.ts:45]: icic ok will refactor the next pr

**Review by ianleck — APPROVED** [review]

**Pull request merged**